pyext/src/analysis_trajectories.py

```diff
@@ -1022,42 +1091,34 @@ def do_extract_models_single_rmf(self, gsms_info, out_rmf_name, traj_dir, analys
         for p in processes:
             p.join()
         
-        output_rmf = analysis_dir + "/" + out_rmf_name
-        scorescat_string = "cat "
+        output_rmf = os.path.join(analysis_dir, out_rmf_name)
+        output_score_file = os.path.join(analysis_dir, scores_prefix+'.txt')
         
-        # Test to see if we have rmf_cat
-        try:
-            subprocess.check_call(["rmf_cat", "-h"], shell=True)
-            rmfcat_string = "rmf_cat "
-            scorescat_string = "cat "
-            # Concatenate the RMF and scorefiles
-            for n in range(len(filenames)):
-                rmfcat_string += filenames[n] +" "
-                scorescat_string += scorefiles[n] +" "
-
-            # output all rmfs to the filename
-            rmfcat_string += output_rmf
-            print("Concatenating",len(filenames)," RMF files to", output_rmf)
+        # use rmf_cat if available
+        if shutil.which('rmf_cat'):
+            # concatenate RMF files
+            rmfcat_string = 'rmf_cat %s %s' % (' '.join(filenames), output_rmf)
+            print('Concatenating', len(filenames), 'RMF files to', output_rmf)
             os.system(rmfcat_string)
-        except subprocess.CalledProcessError:
-            # Otherwise, we need to use RMF
+    
+        # otherwise, fall back on RMF
+        else:
+            print("rmf_cat binary not found on path. Adding frames from output rmf files for each parallel chunk using the RMF library (this will be slow!)")
```

Comment:
This should not be the case... since it looks like below you are doing exactly what `rmf_cat` does anyway. If it is slower, this is a bug and should be fixed.

Reply:
agreed. In commit 871884ee953b0ce282d98cb31373d20bc0b9218e, if ```rmf_cat``` is not found through the ```shutil.which``` search, it raises a ```RuntimeError```